br/tests/utils/start_tls_tikv_cluster_impl

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# --workdir: work directory
+# --tlsdir: certificates directory
+
+set -e
+
+OUT_DIR=
+TLS_DIR=
+API_VERSION=
+
+while [[ ${1} ]]; do
+	case "${1}" in
+	--workdir)
+		OUT_DIR=${2}
+		shift
+		;;
+	--tlsdir)
+		TLS_DIR=${2}
+		shift
+		;;
+	--api_version)
+		API_VERSION=${2}
+		shift
+		;;
+	*)
+		echo "Unknown parameter: ${1}" >&2
+		exit 1
+		;;
+	esac
+
+	if ! shift; then
+		echo 'Missing parameter argument.' >&2
+		exit 1
+	fi
+done
+
+CUR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source $CUR/test_prepare
+
+cd $OUT_DIR && echo "start tikv cluster in $OUT_DIR"
+
+cat - >"$OUT_DIR/pd-config-tls.toml" <<EOF
+[replication]
+# The number of replicas for each region.
+max-replicas = 1
+[security]
+cacert-path = "$TLS_DIR/ca.pem"
+cert-path = "$TLS_DIR/server.pem"
+key-path = "$TLS_DIR/server-key.pem"
+EOF
+
+echo "Starting TLS PD..."
+pd-server \
+	--client-urls https://${TLS_PD_HOST}:${TLS_PD_PORT} \
+	--peer-urls https://${TLS_PD_HOST}:${TLS_PD_PEER_PORT} \
+	--config "$OUT_DIR/pd-config-tls.toml" \
+	--log-file "$OUT_DIR/pd_tls.log" \
+	--data-dir "$OUT_DIR/pd_tls" &
+
+# wait until PD is online...
+while ! curl --cacert $TLS_DIR/ca.pem \
```

Comment:
Here will wait forever if pd-server start failed ? Maybe we need a timeout. Refer to [`timeout`](https://linuxize.com/post/timeout-command-in-linux/) and [here](https://github.com/pingcap/tiup/blob/0be11e93a601d356263c00cd5d67d79b074cc26a/tests/tiup-playground/test_playground.sh#L65).

Reply:
It seems it depend on the http api response to judge whether pd is online and re-call the api in every retry. So I add the `wait_timeout` to let it quit when timeout.